### PR TITLE
Fix reboot

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes

- auto-connect on startup
- port detect on initialization
- cli disconnect
- save and reboot

Tested with auto-connect enabled and disabled.

### CODE REVIEW

File: src/js/serial_backend.js [hunk 1]
```diff
@@ -782,53 +784,45 @@ function startLiveDataRefreshTimer() {
...
+            } else {
+                attempts++;
+                if (attempts > 100) {
+                    clearInterval(reconnect);
+                    console.log(`failed to get serial connection, gave up after 10 seconds`);
+                    gui_log(i18n.getMessage('serialPortOpenFail'));
+                }
             }
         }
     }
```
Comment (McGiverGim) [inline src/js/serial_backend.js:823]: Again, after my PR refactoring: https://github.com/betaflight/betaflight-configurator/pull/3984 there is an easier way to do this: simply put the timestamp into a variable, `rebootTimestamp`, and after receiving the event of new device if this timestamp is small, connect. Easier, without loops, and without polling. I can modify it after both are merged.

**Review by McGiverGim — COMMENTED** [review]
Some comments to not forget to make this way better in a future PR, don't need to change it now because it needs my refactor of the port handler.
But I've tested it, and it seems to not reconnect. I will do more tests and try to report my findings.

File: src/js/serial_backend.js [hunk 2]
```diff
@@ -782,53 +784,24 @@ function startLiveDataRefreshTimer() {
...
-            if (attempts > 100) {
-                clearInterval(reconnect);
-                console.log(`failed to get serial connection, gave up after 10 seconds`);
-                gui_log(i18n.getMessage('serialPortOpenFail'));
-            }
-        }
+    // wait for the device to reboot
+    setTimeout(function() {
+        gui_log(i18n.getMessage('deviceReady'));
+    }, 2000);
```
Comment (McGiverGim) [inline src/js/serial_backend.js:802]: Why this message waiting for 2 seconds? This message must be shown after the device is found again, and, if possible, after connected.

**Review by McGiverGim — APPROVED** [review]
It's ok to me. It does not reconnect after a reboot if you don't have the `auto-connect` enabled, but I can fix this easily later. 
So to me is good to merge this, it's a steep forward.

**Comment by nerdCopter** [comment]
> It's ok to me. It does not reconnect after a reboot if you don't have the `auto-connect` enabled, but I can fix this easily later. So to me is good to merge this, it's a steep forward.

i tested with auto-connect off and on, mine was able to auto-reconnect after save and reboot from Ports tab.  (Chrome / `yarn dev`)

File: src/js/serial_backend.js (hunk 1 above)
Comment (haslinghuis) [reply src/js/serial_backend.js]: Agree, now I moved MSP_SET_REBOOT to reinitialize function timestamp would work better as in the callback it already was updated before comparison.

**Pull request merged**